### ISSUE TITLE
StackOverflow when using DecFP on 32-bit Linux

### ISSUE THREAD

**Issue by omus** [JuliaMath/DecFP.jl#40]
On Ubuntu 16.04 32-bit:

```julia
julia> using DecFP
ERROR: LoadError: StackOverflowError:
while loading /home/omus/.julia/v0.5/DecFP/src/DecFP.jl, in expression starting on line 72
```

**Comment by stevengj** [comment]
That's odd, it works on 32-bit Windows, and as far as I can tell we never use `Int` or `UInt`, only `IntXX` and `UIntXX`.

Can you try deleting lines from DecFP.jl until the error goes away?

**Comment by omus** [comment]
Looks like the problem is [here](https://github.com/stevengj/DecFP.jl/blob/v0.3.0/src/DecFP.jl#L147-L153):
```julia
    for c in (:π, :e, :γ, :catalan, :φ)
        @eval begin
            Base.convert(::Type{$BID}, ::Irrational{$(QuoteNode(c))}) = $(_parse(T, @compat setprecision(256) do
                                                                                      string(BigFloat(eval(c)))
                                                                                  end))
        end
    end
```

**Comment by stevengj** [comment]
Weird... makes me think it is a Julia bug.  Maybe try to do the opposite: construct a minimal test case that exhibits the problem.  (For example, does it occur just for `w==128`?)

**Comment by quinnj** [comment]
Not sure if it's exactly the same issue, but on Julia 1.4.2 32-bit linux, I'm seeing:
```julia
julia> using DecFP
[ Info: Precompiling DecFP [55939f99-70c6-5e9b-8bb0-5071ed7d61fd]

signal (11): Segmentation fault
in expression starting at /root/.julia/packages/DecFP/MVfs9/src/DecFP.jl:281
__bid32_from_string at /root/.julia/artifacts/009db6e3ba7f88193b31b5b2a69da0f8d14edc31/lib/libbid.so (unknown line)
_parse at /root/.julia/packages/DecFP/MVfs9/src/DecFP.jl:290
unknown function (ip: 0xdc80f489)
```

**Comment by stevengj** [comment]
Should be fixed now by #91.

**Comment by omus** [comment]
I can also confirm that #91 fixes the stack overflow issue on Ubuntu 16.04 32-bit using Julia 1.6
